bukkit/src/main/java/io/github/wysohn/triggerreactor/bukkit/main/BukkitPluginManagement.java

```diff
@@ -204,6 +209,48 @@ private ProcessInterrupter.Builder newInterrupterBuilder() {
 
                     return false;
                 })
+                .perExecutor((context, command, args) -> {
+                    if ("GUI".equalsIgnoreCase(command)) {
+                        Future<Boolean> task = taskSupervisor.submitSync(() -> {
+                            String target = null;
+                            String guiName = null;
+
+                            if (args.length == 2 && args[0] instanceof String && args[1] instanceof String) {
+                                target = ((String) args[0]);
+                                guiName = ((String) args[1]);
+                            } else if (args.length == 1 && args[0] instanceof String) {
+
+                                if (context.getVar("player") instanceof Player){
+                                    target = new BukkitPlayer((Player) context.getVar("player")).getName();
+                                } else {
+                                    throw new RuntimeException("Player is null.");
+                                }
+                                guiName = ((String) args[0]);
+
+                            } else {
+                                throw new RuntimeException("Need parameter [String(InventoryTrigger Name)] or [String(Player Name), String(InventoryTrigger Name)]");
+                            }
+
+                            invTriggerManager.openGUI(target, guiName, context.getVarCopy());
+
+                            IInventory inventory = invTriggerManager.openGUI(target, guiName, context.getVarCopy());
+                            if (inventory == null) {
+                                throw new RuntimeException("No such Inventory Trigger named " + guiName);
+                            }
+
+                            return true;
+                        });
+
+                        try {
+                            return task.get();
+                        } catch (InterruptedException | ExecutionException e) {
+                            throw ((RuntimeException) e.getCause());
```

Comment:
You can simply pass the exception as the parameter to the new RuntimeException

`throw new RuntimeException(e)`

Even better if you add a bit of description too

`throw new RuntimeException("something went wrong", e)`

Reply:
Oh I mean change this, and it looks all good

Reply:
> Oh I mean change this, and it looks all good

When i implement it like that, I finally got two error messages

I did it like this because the error message is messy, and it's bad for the aesthetics, should I change it?

Reply:
> > Oh I mean change this, and it looks all good
> 
> When i implement it like that, I finally got two error messages
> 
> I did it like this because the error message is messy, and it's bad for the aesthetics, should I change it?

That's totally okay

And we don't know if the returned value from #getCause() inherits RuntimeException anyway, so I assume those casts would cause cast exception

If you wish, we also could ignore the InterruptedException, which is thrown when the server is shutting down. But this is not a big deal functionality-wise